include/dlaf/multiplication/general.h

```diff
@@ -115,7 +117,18 @@ void generalSubMatrix(comm::CommunicatorGrid grid, const SizeType a, const SizeT
   DLAF_ASSERT(a <= b, a, b);
   DLAF_ASSERT(a >= 0 && a < m, a, m);
 
-  internal::GeneralSub<B, D, T>::callNN(grid, a, b, alpha, mat_a, mat_b, beta, mat_c);
+  internal::GeneralSub<B, D, T>::callNN(row_task_chain, col_task_chain, a, b, alpha, mat_a, mat_b, beta,
+                                        mat_c);
+}
+
+template <Backend B, Device D, class T>
+void generalSubMatrix(comm::CommunicatorGrid grid, const SizeType a, const SizeType b, const T alpha,
+                      Matrix<const T, D>& mat_a, Matrix<const T, D>& mat_b, const T beta,
+                      Matrix<T, D>& mat_c) {
+  common::Pipeline<comm::Communicator> row_task_chain(grid.rowCommunicator());
+  common::Pipeline<comm::Communicator> col_task_chain(grid.colCommunicator());
```

Comment:
I think we should `clone` row/col communicators here, otherwise we would have the pipeline serialising the access to a `Communicator`, which is potentially used somewhere else in parallel (interfering with the serialisation).

Reply:
https://github.com/eth-cscs/DLA-Future/pull/719/commits/0189f92a49066f3374d6a1c0f6580549e87e50f5